### PR TITLE
Feature tour

### PR DESCRIPTION
**Pull request by flozia**
This adds the _What‘s new?_ tour of features that are introduced for the first time. There are two ways for the user to take the _What’s new_ tour:

### Modal alert — Tier 1 ([Figma](https://www.figma.com/file/SjWR5U9aqEiUnmrkuS74O5/What-s-new-Release-notes?node-id=134%3A4832))
If new features were introduced we present the users with a modal containing the tour<sup id="a1">[1](#f1)</sup>. In case the user starts taking the tour all new features are marked as `seen`. In case the modal is dismissed we show an indicator on the settings icon<sup id="a2">[2](#f2)</sup> and _What’s new_ settings menu item<sup id="a3">[3](#f3)</sup>. In either case we present the modal just once every time there are new features.

### Within settings — Tier 2 ([Figma](https://www.figma.com/file/SjWR5U9aqEiUnmrkuS74O5/What-s-new-Release-notes?node-id=134%3A4472))
In the settings there is now a _What’s new_ view<sup id="a4">[4](#f4)</sup> that lists an overview of new features. From here the user has the option to (re-)take the tour<sup id="a5">[5](#f5)</sup>.

### TODO
- [x] Tour popup component
- [x] _What‘s new_ settings view
- [x] Handle presentation to of the _What‘s new_ tour the user
- [x] Add helpers to VPNSettings
- [ ] (Implement WhatsNewModel)

### Screenshots
<b id="f1">1</b> Modal alert [↩](#a1)
tour-initial-popup

<b id="f2">2</b> Indicator on settings icon [↩](#a2)
tour-settings-indicator

<b id="f3">3</b> Indicator in settings [↩](#a3)
tour-settings-item-indicator

<b id="f4">4</b> What’new overview [↩](#a4)
tour-features-list

<b id="f5">5</b> (Re-)take tour in settings [↩](#a5)
tour-opened-from-settings

### CODE REVIEW

File: src/settingsholder.cpp [hunk 1]
```diff
@@ -177,6 +183,8 @@ void SettingsHolder::clear() {
   m_settings.remove(SETTINGS_SURVEYS);
   m_settings.remove(SETTINGS_IAPPRODUCTS);
   m_settings.remove(SETTINGS_POSTAUTHENTICATIONSHOWN);
+  m_settings.remove(SETTINGS_NEWFEATURESSEEN);
+  m_settings.remove(SETTINGS_FEATURESTOURSHOWN);
```
Comment (bakulf) [inline src/settingsholder.cpp:187]: This means that after a logout, the user will see all the new features again. Is it OK?

File: src/ui/settings/ViewSettingsMenu.qml [hunk 2]
```diff
@@ -105,7 +105,19 @@ VPNFlickable {
...
         width: parent.width - Theme.windowMargin
         anchors.horizontalCenter: parent.horizontalCenter
+        VPNSettingsItem {
+            objectName: "settingsWhatsNew"
+
+            settingTitle: "What’s new"
+            imageLeftSrc: "../resources/gift-dark.svg"
+            imageRightSrc: "../resources/chevron.svg"
+            onClicked: settingsStackView.push("../settings/ViewWhatsNew.qml")
+            showIndicator: !VPNSettings.newFeaturesSeen
```
Comment (bakulf) [inline src/ui/settings/ViewSettingsMenu.qml:115]: What about new releases and new features? We can do better than this! What about if you check all the feature names marked as new (`isNew`), and you compare them with an `QStringList` of seen-features stored in settings?

File: src/settingsholder.cpp (hunk 1 above)
Comment (flozia) [reply src/settingsholder.cpp:187]: Thank you for catching this. No it’s not OK and I’ll remove this.

File: src/models/feature.h [hunk 3]
```diff
@@ -28,9 +29,9 @@ class Feature : public QObject {
   // protected:
  public:
   Feature(QString id, L18nStrings::String name_id, bool isMajor,
-          L18nStrings::String shortDesc_id, L18nStrings::String desc_id,
-          QString imgPath, QString iconPath, QString releaseVersion,
-          bool devModeWriteable = false);
+          L18nStrings::String displayName_id, L18nStrings::String shortDesc_id,
```
Comment (bakulf) [inline src/models/feature.h:32]: This patch needs to be rebased :/

File: src/ui/components/VPNPopup.qml [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    topPadding: Theme.popupMargin * 2
+
+    background: Rectangle {
+        id: popupBackground
+
+        anchors.margins: 0
+        color: Theme.bgColor
+        radius: 8
+
+        DropShadow {
```
[the remark this answers is not in the document]
Comment (flozia) [reply src/ui/components/VPNPopup.qml:31]: Thanks for saving me from debugging this — I applied your fix. Just a quick note: It seems like setting `color: "transparent"` prevents drawing the shadow.

File: src/ui/main.qml [hunk 5]
```diff
@@ -320,13 +320,74 @@ Window {
...
+
+    VPNFilterProxyModel {
+        id: newFeaturesModel
+        source: VPNFeatureList
+        // Filter features that should be listed in What’s new
+        filterCallback: feature => showFeatureInWhatsNew(feature)
+    }
+
+    VPNFilterProxyModel {
+        id: unseenFeaturesModel
```
[the remark this answers is not in the document]
Comment (flozia) [reply src/ui/main.qml:350]: :)

**Pull request merged**